eqcorrscan/utils/catalog_to_dd.py

```diff
@@ -135,10 +140,10 @@ def _generate_event_id_mapper(catalog, event_id_mapper=None):
     except ValueError:
         largest_event_id = 0
     for event in catalog:
-        if event.resource_id.id not in event_id_mapper.keys():
+        if str(event.resource_id) not in event_id_mapper.keys():
             event_id = largest_event_id + 1
             largest_event_id = event_id
-            event_id_mapper.update({event.resource_id.id: event_id})
+            event_id_mapper.update({str(event.resource_id): event_id})
```

Comment:
Was there a reason to change this? The `.__str__` method on `ResourceIdentifier` returns the `.id` property anyway (see [here](https://github.com/obspy/obspy/blob/master/obspy/core/event/resourceid.py#L672)) so it shouldn't make a difference which is called?

Reply:
By using `str()`, the function can take as an argument both the `SparseEvent` and the normal `Event`. With `event.resource_id.id`, it is only compatible with `obspy.core.event.Event`.

Reply:
Nice!